src/main/java/net/glowstone/util/config/ServerConfig.java

```diff
@@ -393,6 +393,12 @@ private boolean migrate() {
                     .add(new Library("org.apache.commons", "commons-lang3", "3.5",
                             HashAlgorithm.SHA1, "6c6c702c89bfff3cd9e80b04d668c5e190d588c6")
                             .toConfigMap())
+                    .add(new Library("org.jetbrains.kotlin", "kotlin-runtime", "1.2.21",
```

Comment:
Could you extract the Kotlin version to a constant? Makes sure the versions are kept in sync when updating.

Reply:
I can do that, but one thing to keep in mind is that updating the version in our code won't change the version in existing configs. Only new configs will inherit this version.

Reply:
I find this to be a regression. Before, server owners were automatically updated to the latest lib version but now it must be a manual change.

Reply:
Mastercoms, by "this", do you meant all libraries that we moved into the config? Or are you talking specifically about moving Kotlin out of the POM and into the configs?

Reply:
All the libraries. I think we should have two configuration sections: one for "builtin" libraries that Glowstone downloads by default which should only allow for a simple toggle to download them or not, and one for custom libraries that the server owner can specify.

Reply:
I would argue that if these libraries should be upgraded with a server upgrade, they should be embedded into the JAR. Otherwise, they are optional, only commonly required by plugins that the server operator may run, and should be under the server operator's control. With the "builtin" libraries section as you suggest, it'd be impossible for a server operator to upgrade or downgrade a library when a plugin depends on a specific version, especially with the SHAs now in place.

Reply:
How about we add a warning when a version of the downloaded library is lower than the one in the "builtin" library map?

Reply:
These builtin libraries, excluding the Kotlin libs, are part of a CraftBukkit environment. We should be emulating that environment and thus we should have control over what that environment is.

Arguably, having a up to date and universal Kotlin environment controlled by us is also beneficial but I would be fine with just a warning about the Kotlin libs being out of date, as momo suggested.